src/services/options/network-admin-options-service.php

```diff
@@ -0,0 +1,210 @@
+
+	/**
+	 * Holds the multisite' network option configurations.
+	 *
+	 * {@inheritDoc}
+	 *
+	 * @var array[string]
+	 */
+	protected $configurations = [
+		'access'      => [
+			'default'    => 'admin',
+			'types'      => [
+				'in_array' => [
+					'allow' => [
+						'admin',
+						'superadmin',
+					],
+				],
+			],
+			'ms_exclude' => false,
+		],
+		'defaultblog' => [
+			'default' => '',
+			'types'   => [ 'empty_string', 'integer' ],
+		],
+	];
+
+	/**
+	 * Holds the multisite options service instance.
+	 *
+	 * @var Multisite_Options_Service
+	 */
+	protected $multisite_options_service;
+
+	/**
+	 * Sets the dependencies.
+	 *
+	 * Use this for dependencies that are not included via the Abstract_Options_Service's constructor.
+	 *
+	 * @param Multisite_Options_Service $multisite_options_service The multisite options service.
+	 *
+	 * @required
+	 */
+	public function set_dependencies( Multisite_Options_Service $multisite_options_service ) {
+		$this->multisite_options_service = $multisite_options_service;
+	}
+
+	/**
+	 * Resets all options for the current blog when the `ms_defaults_set` option is false.
+	 *
+	 * @throws Save_Failed_Exception When the save failed.
+	 * @throws Delete_Failed_Exception When the deletion failed.
+	 * @throws Unknown_Exception When the `defaultblog` option is not registered.
+	 *
+	 * @return bool Whether the options were reset.
+	 */
+	public function maybe_reset_current_blog_options() {
+		try {
+			$is_ms_defaults_set = $this->multisite_options_service->__get( 'ms_defaults_set' );
+		} catch ( Unknown_Exception $e ) {
+			$is_ms_defaults_set = false;
+		}
+
+		if ( $is_ms_defaults_set ) {
+			return false;
+		}
+		$this->reset_options_for( \get_current_blog_id() );
+
+		return true;
+	}
+
+	// phpcs:disable Squiz.Commenting.FunctionCommentThrowTag.WrongNumber -- PHPCS doesn't take into account exceptions thrown in called methods.
+
+	/**
+	 * Resets all options for a specific multisite blog.
+	 *
+	 * The new options are retrieved from the specified default blog, if one was chosen on the network page, or the
+	 * plugin defaults if it was not.
+	 *
+	 * @param int $blog_id The blog ID.
+	 *
+	 * @throws Save_Failed_Exception When the save failed.
+	 * @throws Delete_Failed_Exception When the deletion failed.
+	 * @throws Unknown_Exception When the `defaultblog` option is not registered.
+	 *
+	 * @return void
+	 */
+	public function reset_options_for( $blog_id ) {
+		// Determine the blog to get the options from.
+		$base_blog_id = $blog_id;
+		// Use the default blog when that is set.
+		$default_blog = $this->__get( 'defaultblog' );
+		if ( $default_blog !== '' && $default_blog !== 0 ) {
+			$base_blog_id = $default_blog;
+		}
+
+		// Delete the old options.
+		if ( ! \delete_blog_option( $blog_id, $this->multisite_options_service->option_name ) ) {
+			throw Delete_Failed_Exception::for_option( $this->multisite_options_service->option_name );
+		}
```

Comment:
Suggestion to first check if the option exist else it will fatal on a clean WordPress installation. 
```suggestion
		if ( \get_blog_option($blog_id, $this->multisite_options_service->option_name) ) {
			if ( ! \delete_blog_option( $blog_id, $this->multisite_options_service->option_name ) ) {
				throw Delete_Failed_Exception::for_option( $this->multisite_options_service->option_name );
			}
		}
```